hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -219,20 +219,22 @@ struct Vgen {
     , catches(env.catches)
   {}
   ~Vgen() {
-    auto begin = reinterpret_cast<char*>(base);
-    auto end = reinterpret_cast<char*>(a->frontier());
-    __builtin___clear_cache(begin, end);
+    auto begin = env.cb->toDestAddress(base);
+    auto end = env.cb->toDestAddress(a->frontier());
+    __builtin___clear_cache(reinterpret_cast<char*>(begin),
+                            reinterpret_cast<char*>(end));
```

Comment:
Maybe a helper that takes a CodeBlock, and begin and end addresses and clears the cache is in order...